packages/devextreme/js/ui/menu/ui.menu.js

```diff
@@ -674,13 +684,36 @@ class Menu extends MenuBase {
         }
     }
 
-    _submenuOnHiddenHandler($rootItem, submenu) {
+    _submenuOnHiddenHandler($rootItem, submenu, eventArgs) {
+        const { itemData, $submenuContainer } = this._getVisibilityChangeEventParams(eventArgs.rootItem, submenu, $rootItem, true);
+
         this._actions.onSubmenuHidden({
+            itemData,
             rootItem: getPublicElement($rootItem),
-            submenu: submenu
+            submenuContainer: getPublicElement($submenuContainer),
+            submenu,
         });
     }
 
+    _getVisibilityChangeEventParams(submenuItem, submenu, $rootItem, isHide) {
+        let itemData;
+        let $submenuContainer;
+
+        if(submenuItem) {
+            const anchor = isHide ? $(submenuItem).closest(`.${DX_MENU_ITEM_CLASS}`)[0] : submenuItem;
```

Comment:
So, `submenuItem` has different value on showing and on hiding? Looks incorrect.
Cannot we change it? Or why is it necessary?

Reply:
Yes, we have different values in show and hide events. For show this is the element which we want to expand. For hide - submenu which we are closing. And there is a logic in this.